src/main/java/org/folio/ebsconet/mapper/OrdersMapper.java

```diff
@@ -172,18 +186,18 @@ else if ((poLine.getCost().getQuantityElectronic() > 1 && poLine.getCost().getQu
 
       poLine.getCost().setQuantityElectronic(newElectronicQuantity);
       poLine.getCost().setQuantityPhysical(newPhysicalQuantity);
-      poLine.getLocations().get(0).setQuantityElectronic(newElectronicQuantity);
-      poLine.getLocations().get(0).setQuantityPhysical(newPhysicalQuantity);
+
+      if (CollectionUtils.isNotEmpty(poLine.getLocations())) {
+        poLine.getLocations().get(0).setQuantityElectronic(newElectronicQuantity);
+        poLine.getLocations().get(0).setQuantityPhysical(newPhysicalQuantity);
+      }
     }
   }
 
   private void redistributeSinglePeMixItem(CompositePoLine poLine) {
     poLine.getCost().setQuantityElectronic(1);
     poLine.getCost().setQuantityPhysical(1);
 
-    // set all location quantities to zero
-    clearLocationQuantities(poLine);
```

Comment:
Is it possible if we have for example 3 locations in composite po line, so now after we deleting clearLocationQuantities() the third one will contain always old values?

Reply:
I see now, this method invoked only from the single place where clearLocationQuantities90 invoked, comment is resolved